build.sh

```diff
@@ -390,6 +490,179 @@ if [ "$DEBUG" = true ] ; then
+    exit 1
+fi
+
+if [ "$DEBUG" = true ] ; then
+    echo "PYTHON: $PYTHON"
+fi
+
+################################################################################
+# Build PKI
+################################################################################
+
+if [ "$BUILD_TARGET" = "dist" ] ; then
+
+    if [ "$VERBOSE" = true ] ; then
+        echo "Building $NAME"
+    fi
+
+    OPTIONS=()
+
+    OPTIONS+=(-S $SRC_DIR)
+    OPTIONS+=(-B $WORK_DIR)
+
+    # Set environment variables for CMake
+    # (see /usr/lib/rpm/macros.d/macros.cmake)
+
+    OPTIONS+=(-DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG)
+    OPTIONS+=(-DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG)
+    OPTIONS+=(-DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG)
+    OPTIONS+=(-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON)
+    OPTIONS+=(-DCMAKE_INSTALL_DO_STRIP:BOOL=OFF)
+    OPTIONS+=(-DCMAKE_INSTALL_PREFIX:PATH=$PREFIX_DIR)
+
+    OPTIONS+=(-DINCLUDE_INSTALL_DIR:PATH=$INCLUDE_DIR)
+    OPTIONS+=(-DLIB_INSTALL_DIR:PATH=$LIB_DIR)
+    OPTIONS+=(-DSYSCONF_INSTALL_DIR:PATH=$SYSCONF_DIR)
+    OPTIONS+=(-DSHARE_INSTALL_PREFIX:PATH=$SHARE_DIR)
+
+    OPTIONS+=(-DLIB_SUFFIX=64)
+    OPTIONS+=(-DBUILD_SHARED_LIBS:BOOL=ON)
+
+    OPTIONS+=(--no-warn-unused-cli)
+    OPTIONS+=(-DPRODUCT_NAME="$PRODUCT_NAME")
+    OPTIONS+=(-DTHEME=$THEME)
+    OPTIONS+=(-DVERSION=$VERSION)
+    OPTIONS+=(-DRELEASE=$RELEASE)
+
+    OPTIONS+=(-DVAR_INSTALL_DIR:PATH=/var)
+    OPTIONS+=(-DP11_KIT_TRUST=$P11_KIT_TRUST)
+
+    if [ "$JAVA_HOME" != "" ] ; then
+        OPTIONS+=(-DJAVA_HOME=$JAVA_HOME)
+    fi
+
+    OPTIONS+=(-DJAVA_LIB_INSTALL_DIR=$JNI_DIR)
+    OPTIONS+=(-DAPP_SERVER=$APP_SERVER)
+    OPTIONS+=(-DPYTHON_EXECUTABLE=$PYTHON)
+    OPTIONS+=(-DSYSTEMD_LIB_INSTALL_DIR=$UNIT_DIR)
+
+    for package in ${PKGS_TO_SKIP[@]}
+    do
+        package=${package^^}
+        OPTIONS+=(-DWITH_$package:BOOL=OFF)
+    done
+
+    if [ "$WITH_CONSOLE" = true ] ; then
+        OPTIONS+=(-DWITH_CONSOLE:BOOL=ON)
+    fi
+
+    if [ "$WITHOUT_TEST" = true ] ; then
+        OPTIONS+=(-DRUN_TESTS:BOOL=OFF)
+    fi
+
+    $CMAKE "${OPTIONS[@]}"
+
+    OPTIONS=()
+
+    if [ "$VERBOSE" = true ] ; then
+        OPTIONS+=(VERBOSE=1)
+    fi
+
+    OPTIONS+=(CMAKE_NO_VERBOSE=1)
+    OPTIONS+=(--no-print-directory)
+
+    make "${OPTIONS[@]}" all
+
+    if [ "$WITH_TESTS" = true ] ; then
+        ctest --output-on-failure
+    fi
+
+    echo
+    echo "Build artifacts:"
+    echo "- Java archives:"
+    echo "    $WORK_DIR/dist/pki-cmsutil.jar"
+    echo "    $WORK_DIR/dist/pki-certsrv.jar"
+    echo "    $WORK_DIR/dist/pki-tools.jar"
+    echo "    $WORK_DIR/dist/pki-tomcat.jar"
+    echo "    $WORK_DIR/dist/pki-cms.jar"
+    echo "    $WORK_DIR/dist/pki-cmsbundle.jar"
+    echo "    $WORK_DIR/dist/pki-acme.jar"
+    echo "    $WORK_DIR/dist/pki-ca.jar"
+    echo "    $WORK_DIR/dist/pki-kra.jar"
+    echo "    $WORK_DIR/dist/pki-ocsp.jar"
+    echo "    $WORK_DIR/dist/pki-tks.jar"
+    echo "    $WORK_DIR/dist/pki-tps.jar"
```

Comment:
Are these always built or they depend on the package list?

Reply:
Good point. They depend on the package list. I've updated the code.